.github/workflows/on-pull-request.yml

```diff
@@ -13,20 +13,41 @@ env:
   PASSAGE_AUTH_TOKEN: ${{ secrets.PASSAGE_AUTH_TOKEN }}
 
 jobs:
-  build:
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version: '1.20.14'
+      - name: Lint
+        uses: golangci/golangci-lint-action@v6
+        with:
+          version: v1.60
+  format:
+    name: Format
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version: '1.20.14'
+      - name: Format
+        run: |
+          if [[ $(gofmt -s -d .) ]]; then
+            echo "Run 'gofmt -s -w .' to format code."
```

Comment:
considered posting a PR comment for this and the linter failure results as a convenience for contributors, but didn't want to overcomplicate the workflow